File: solutions/javascript/word-count/1/word-count.js
export default class Words {
  count(phrase) {
    const freq = {};
    phrase.toLocaleLowerCase()
      .trim()
      .split(/\s+/)
      .forEach((word) => {
        const count = freq[word] || 0;
        freq[`${word}`] = count + 1;
      });
    return freq;
  }
}

File: solutions/elixir/word-count/1/word_count.exs
defmodule Words do
  @doc """
  Count the number of words in the sentence.

  Words are compared case-insensitively.
  """
  @spec count(String.t) :: map
  def count(sentence) do
    sentence
    |> String.replace(~r{[,:!&@$%^&_]}u, " ")  # [[:punct:]] includes -
    |> String.split()
    |> Enum.group_by(&(String.downcase(&1)))
    |> Enum.map(fn {k, v} -> {k, length(v)} end)
    |> Enum.into(%{})
  end
end